File: ezSCORES/ezSCORES.Model/DTOs/MatchDTO.cs
﻿using ezSCORES.Model.ENUMs;
using System;
using System.Collections.Generic;
using System.Text;

namespace ezSCORES.Model.DTOs
{
	public class MatchDTO
	{
		public int MatchId { get; set; }
		public int? HomeTeamScore { get; set; }
		public int? AwayTeamScore { get; set; }
		public DateTime DateAndTime { get; set; }
		public TeamDTO HomeTeam { get; set; }
		public TeamDTO AwayTeam { get; set; }
		public Stadiums Stadium { get; set; }
		public List<GoalDTO> Goals { get; set; }
		public List<RefereesDTO> Referees { get; set; }
		public int FixtureId { get; set; }
		public GroupDTO? Group { get; set; }
		public int FixtureSequenceNumber { get; set; }
		public GameStage GameStage{ get; set; }
		public bool IsUnderway { get; set; }
		public bool IsCompleted { get; set; }
	}
}

File: ezSCORES/UI/ezscores_mobile/lib/models/DTOs/matchDto.dart
import 'package:ezscores_mobile/models/DTOs/goalDto.dart';
import 'package:ezscores_mobile/models/DTOs/groupDto.dart';
import 'package:ezscores_mobile/models/DTOs/refereeDto.dart';
import 'package:ezscores_mobile/models/DTOs/teamDto.dart';
import 'package:ezscores_mobile/models/enums/gameStage.dart';
import 'package:ezscores_mobile/models/enums/game_stage_converter.dart';
import 'package:ezscores_mobile/models/stadiums.dart';
import 'package:json_annotation/json_annotation.dart';

part 'matchDto.g.dart';

@JsonSerializable()
class MatchDTO {
	int? matchId;
	int? homeTeamScore;
	int? awayTeamScore;
	DateTime? dateAndTime;
  TeamDTO? homeTeam;
  TeamDTO? awayTeam;
	Stadiums? stadium;
  List<GoalDTO>? goals;
	List<RefereeDTO>? referees;
	int? fixtureId;
	GroupDTO? group;
	int? fixtureSequenceNumber;

  @GameStageConverter()
  GameStage? gameStage;
  
	bool? isUnderway;
	bool? isCompleted;

	MatchDTO();

	factory MatchDTO.fromJson(Map<String, dynamic> json) => _$MatchDTOFromJson(json);

	Map<String, dynamic> toJson() => _$MatchDTOToJson(this);
}